src/tests/factory_dependency.rs

```diff
@@ -7,11 +7,37 @@
 use std::collections::BTreeMap;
 
 use crate::solc::pipeline::Pipeline as SolcPipeline;
+use crate::solc::Compiler as SolcCompiler;
+
+#[test]
```

Comment:
@hedgar2017 ,

Should we think about using [`rstest` and parametrized test cases](https://crates.io/crates/rstest)? It should remove all this boilerplate code.

<img width="591" alt="image" src="https://github.com/matter-labs/era-compiler-solidity/assets/22070213/0bb891ff-3353-4459-a8d9-4bb24358e376">

Reply:
Yes, let's try this in another PR.
I'm wary about macro solutions in Rust, as they're basically developed from scratch and you may run into sort of JS-style DevEx :)